biosys/apps/main/tests/api/test_species_observation.py

```diff
@@ -272,137 +269,143 @@ def test_options(self):
-        self.custodian_2_user.save()
-        self.custodian_2_client = APIClient()
-        self.assertTrue(self.custodian_2_client.login(username=self.custodian_2_user.username, password=password))
-        self.project_2 = Project.objects.filter(name="Project2").first()
-        self.site_2 = Site.objects.filter(code="Site2").first()
-        self.ds_2 = Dataset.objects.filter(name="Bats2", project=self.project_2).first()
-        self.assertTrue(self.ds_2.is_custodian(self.custodian_2_user))
-        self.assertFalse(self.ds_1.is_custodian(self.custodian_2_user))
-
-        self.readonly_user = User.objects.filter(username="readonly").first()
-        self.assertIsNotNone(self.custodian_2_user)
-        self.assertFalse(self.site_2.is_custodian(self.readonly_user))
-        self.assertFalse(self.site_1.is_custodian(self.readonly_user))
-        self.readonly_user.set_password(password)
-        self.readonly_user.save()
-        self.readonly_client = APIClient()
-        self.assertTrue(self.readonly_client.login(username=self.readonly_user.username, password=password))
+        project = self.project_1
+        client = self.custodian_1_client
+        schema = self.schema_with_species_name()
+        self.ds_1 = self._create_dataset_with_schema(project, client, schema, dataset_type=Dataset.TYPE_SPECIES_OBSERVATION)
 
-        self.anonymous_client = APIClient()
+    def _create_default_record(self):
+        ds = self.ds_1
+        client = self.custodian_1_client
+        data = {
+            'Species Name': 'Chubby Bat',
+            'Latitude': -32.0,
+            'Longitude': 115.75,
+            'When': '2018-01-31'
+        }
+        payload = {
+            "dataset": ds.pk,
+            "data": data
+        }
+        url = reverse('api:record-list')
+        ds.record_queryset.delete()
+        self.assertEqual(
+            client.post(url, data=payload, format='json').status_code,
+            status.HTTP_201_CREATED
+        )
+        return ds.record_queryset.first()
 
     def test_create_one_happy_path(self):
         """
         Test the create of one record
         :return:
         """
-        # grab one existing an re-inject it
-        record = self.record_1
         ds = self.ds_1
+        client = self.custodian_1_client
         data = {
-            "dataset": record.dataset.pk,
-            "data": record.data
+            'Species Name': 'Chubby Bat',
+            'Latitude': -32.0,
+            'Longitude': 115.75,
+            'When': '2018-01-31'
+        }
+        payload = {
+            "dataset": ds.pk,
+            "data": data
         }
         url = reverse('api:record-list')
-        client = self.custodian_1_client
-        count = ds.record_queryset.count()
+        ds.record_queryset.delete()
         self.assertEqual(
-            client.post(url, data, format='json').status_code,
+            client.post(url, data=payload, format='json').status_code,
```

Comment:
not really necessary to use keyword arg here

Reply:
yep